data_override/include/data_override.inc

```diff
@@ -591,9 +591,18 @@ subroutine read_table_yaml(data_table)
     integer :: nentries, mentries
     integer :: i
     character(len=50) :: buffer
+    character(len=255) :: filename !< Name of the expected data_table.yaml
     integer :: file_id
 
-    file_id = open_and_parse_file("data_table.yaml")
+    ! If doing and ensemble or nest run add the filename appendix (ens_XX or nest_XX) to the filename
+    call get_instance_filename("data_table.yaml", filename)
+    if (index(trim(filename), "ens_") .ne. 0) then
+      if (file_exists(filename) .and. file_exists("data_table.yaml")) &
+        call mpp_error(FATAL, "Both data_table.yaml and "//trim(filename)//" exists, pick one!")
+    endif
+
+    file_id = open_and_parse_file(trim(filename))
+
```

Comment:
Need to make it clear in any documentation we are only testing for collisions with `ens_xx` and a generic data_table.yaml and not for `nest_xx` as well.